articles/server-apis/webapi-owin/02-authorization.md

```diff
@@ -0,0 +1,120 @@
+    'Auth0.OpenIdConnectSigningKeyResolver NuGet Package v1.0.0'
+  ]
+}) %>
+
+An API can enforce fine grained control over who can access the various endpoints exposed by the API. These permissions are expressed as scopes.
+
+When a user authorizes a client application, the application can also indicate which permissions it requires. The user is allowed to review and grant these permissions. These permissions are then included in the `access_token` as part of the `scope` claim.
+
+Subsequently when the client passes along the `access_token` when making requests to the API, the API can query the `scope` claim to ensure that the required permissions were granted in order to call the particular API endpoint. The `scope` claim will contain the list of scopes separated by a space, as can be seen in the example JWT payload below:
+
+```json
+{
+  "iss": "https://auth0pnp.auth0.com/",
+  "sub": "5fJ...",
+  "aud": "https://api.myapp.com/timesheets",
+  "exp": 1479957385,
+  "iat": 1479870985,
+  "scope": "delete:timesheets update:timesheets create:timesheets read:timesheets"
+}
+```
+
+## 1. Define scopes
+
+You will need to define to list of scopes available in your application. In the [APIs section]("${manage_url}/#/apis) of the Auth0 Dashboard, select your API and then go to the __Scopes__ tab of the API. Configure the list of Scopes available in your API:
+
+![Configure Scopes](/media/articles/server-apis/webapi-owin/create-api-scopes.png)
+
+Once you have created the list of scopes, go to the __Non Interactive Clients__ tab. For each client which is able to access your API, ensure that you have Authorized the client and selected the Scopes which can be granted to the Client. Be sure the click the **Update** button once you have configure the Client:
+
+![Configure Client](/media/articles/server-apis/webapi-scopes/configure-api-client-scopes.png)
+
+## 2. Enforcing scopes in your ASP.NET Web API 
+
+To ensure that a correct `scope` is present in order to execute a particular API endpoint, you can create a custom Authorization Attribute. 
+
+Create a class called `ScopeAuthorizeAttribute` which inherits from `System.Web.Http.AuthorizeAttribute`. This Authorization Attribute will check that the `scope` claim issued by your Auth0 tenant is present, and if so it will ensure that the `scope` claim contains the requested scope.
+
+```csharp
+public class ScopeAuthorizeAttribute : AuthorizeAttribute
+{
+    private readonly string scope;
+
+    public ScopeAuthorizeAttribute(string scope)
+    {
+        this.scope = scope;
+    }
+    public override void OnAuthorization(HttpActionContext actionContext)
+    {
+        base.OnAuthorization(actionContext);
+
+        ClaimsPrincipal principal = actionContext.ControllerContext.RequestContext.Principal as ClaimsPrincipal;
+        if (principal != null)
+        {
+            // If user does not have the scope claim, get out of here
+            if (principal.HasClaim(c => c.Type == "scope"))
```

Comment:
We should. Thing is however, how do you pass it in to the Attribute? 

You can only pass in a constant value as a parameter to an Attribute, which means the developers cannot configure the Issuer in the configuration file, and will instead have to hard code it at compile time.

I am open to suggestions to work around this problem. Source code is at https://github.com/auth0-samples/auth0-aspnet-owin-webapi-sample/tree/master/02-Authorization/WebApi if you want to play around with it :)